lua/telescope/_extensions/gh_actions.lua

```diff
@@ -143,54 +144,75 @@ A.gh_run_rerun=function(prompt_bufnr)
   os.execute('gh run rerun ' .. selection.id)
 end
 
-A.gh_run_view_log=function(prompt_bufnr)
-  local selection = action_state.get_selected_entry(prompt_bufnr)
-  actions.close(prompt_bufnr)
-  if selection.id == "" then
-    return
-  end
-  local log_output = {}
-  vim.api.nvim_command('botright vnew')
-  local buf = vim.api.nvim_get_current_buf()
-
-  vim.api.nvim_buf_set_name(0, 'result #' .. buf)
-
-  vim.api.nvim_buf_set_option(0, 'buftype', 'nofile')
-  vim.api.nvim_buf_set_option(0, 'swapfile', false)
-  vim.api.nvim_buf_set_option(0, 'filetype', 'result')
-  vim.api.nvim_buf_set_option(0, 'bufhidden', 'wipe')
-
-  vim.api.nvim_command('setlocal wrap')
-  vim.api.nvim_command('setlocal cursorline')
-
-  local on_output = function(_, line)
-    table.insert(log_output,line)
-    pcall(vim.schedule_wrap( function()
-      vim.api.nvim_buf_set_lines(buf, 0, -1, false, log_output)
-     end))
-  end
-
-  local job = Job:new({
-      enable_recording = true ,
-      command = "gh",
-      args = flatten{"run", "view" , "--log", selection.id},
-      on_stdout = on_output,
-      on_stderr = on_output,
+A.gh_run_view_log=function(opts)
+  return function(prompt_bufnr)
+    local selection = action_state.get_selected_entry(prompt_bufnr)
+    actions.close(prompt_bufnr)
+    if selection.id == "" then
+      return
+    end
+    local log_output = {}
+    vim.api.nvim_command(opts.wincmd)
+    local buf = vim.api.nvim_get_current_buf()
+
+    vim.api.nvim_buf_set_name(0, 'result #' .. buf)
+
+    vim.api.nvim_buf_set_option(0, 'buftype', 'nofile')
+    vim.api.nvim_buf_set_option(0, 'swapfile', false)
+    vim.api.nvim_buf_set_option(0, 'filetype', 'result')
+    vim.api.nvim_buf_set_option(0, 'bufhidden', 'wipe')
+    vim.api.nvim_command('setlocal ' .. opts.wrap)
+    vim.api.nvim_command('setlocal cursorline')
+
+    local args = {}
+    local run_completed = false
+    if selection.status == "success" or selection.status == "failure" then
+      run_completed = true
+    end
+    if run_completed then
+      args = {"run", "view" , "--log", selection.id}
+    else
+      args = {"run", "watch", selection.id}
+    end
 
-      on_exit = function(_,status)
-        if status == 0 then
-           print("Log retrieval completed!")
+    vim.api.nvim_buf_set_lines(buf, 0, -1, false, {"Retrieving log, please wait..."})
+
+    local on_output = function(_, line)
+      if not run_completed and string.find(line, "Refreshing") then
+        log_output = {}
+      end
+      if opts.cleanmeta and run_completed then
+        -- Lua doesn't support capture group quantifier
+        local tmp = string.gsub(line, "(.*\t)", "")
+        line = string.gsub(tmp, "(.*%d+Z)", "")
```

Comment:
this will delete all text with color hightlight you can test on telescope main repo. 

```lua
        line = string.gsub(line, "^(.*%d%d%d%d%d%d%dZ)", "")
```
we should delete only text form start to time Z.

Reply:
Thanks for pointing out. I don't use colors, so I totally missed that. I was able to reproduce the issue, so when I have some time Ill fix it

Reply:
Hi, missing log lines should be fixed now. I can see color highlight in cleared output, for example:
``` 
[36;1mecho "::set-output name=id::$(date +%Y%m%d-%H%M%S)"[0m
```